base/sparse/linalg.jl

```diff
@@ -879,7 +879,7 @@ for f in (:\, :Ac_ldiv_B, :At_ldiv_B)
             if m == n
                 if istril(A)
                     if istriu(A)
-                        return ($f)(Diagonal(A), B)
+                        return ($f)(Diagonal(Vector(diag(A))), B)
```

Comment:
Is this to avoid changing the behavior? I guess it should still work with a sparse vector, right?